### PR TITLE
feat: add conversation list cubit

### PR DESCRIPTION
**Pull request by boxdot**
The conversation list cubit contains the logic for loading conversations. Its state is a list of conversations. The conversation list view is now a stateless widget that uses this cubit.

Note: A conversation is not automatically opened on desktop now. Most likely the currently opened conversation should be stored in the local storage. Automatically opening a conversation is not consistent between the desktop and mobile clients.

Also fix: too eager caching of conversation avatars.

A common implementation with a background emitter loop task was extracted in to the `CubitCore` struct.

### CODE REVIEW

File: applogic/src/util/cubit_core.rs [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+                    trace!("emitting new state, sinks = {}, state = {:?}", sinks.len(), state);
+                    sinks.retain(|sink| sink.add(state.clone()).is_ok());
+                },
+                _ = stop.cancelled() => {
+                    return;
+                }
+            }
+        }
+    }
+}
```
Comment (raphaelrobert) [inline applogic/src/util/cubit_core.rs:117]: If this is something that's not just temporary, can you add some comments?

File: applogic/src/api/conversation_list_cubit.rs [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+use tokio::sync::{broadcast, watch};
+use tokio_util::sync::CancellationToken;
+
+use crate::api::user::User;
+use crate::util::{spawn_from_sync, Cubit, CubitCore};
+use crate::StreamSink;
+
+use super::messages::{FetchedMessages, FetchedMessagesReceiver};
+use super::types::UiConversationDetails;
+use super::user::user_cubit::UserCubitBase;
```
Comment (raphaelrobert) [inline applogic/src/api/conversation_list_cubit.rs:18]: Is there a good reason not to merge the imports? We should agree on a standard. So far we have used merged imports, but I'm open to good arguments.

**Review by raphaelrobert — COMMENTED** [review]
Thanks, some comments for now.

File: applogic/src/util/cubit_core.rs (hunk 1 above)
Comment (boxdot) [reply applogic/src/util/cubit_core.rs:147]: Good point. It is definitely under-documented. Added docs.

File: applogic/src/api/conversation_list_cubit.rs (hunk 2 above)
Comment (boxdot) [reply applogic/src/api/conversation_list_cubit.rs:18]: No particular reason except my global config is like that:

```lua
imports = {
    granularity = {
        group = "module",
    },
},
```

Most likely we should configure this in `.rustfmt.toml`

Comment (boxdot) [reply applogic/src/api/conversation_list_cubit.rs:18]: Unfortunately, https://rust-lang.github.io/rustfmt/?version=v1.6.0&search=imports#imports_granularity is not yet stable.

**Pull request merged**